### PR TITLE
libmesh: added mpi variant

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by aprokop** [comment]
I turned it on by default but not 100% sure if that's the right thing.

**Comment by aprokop** [comment]
This is WIP right now, as I am hitting
```
[0] ./include/libmesh/eigen_sparse_vector.h, line 506, compiled Dec 13 2016 at 20:12:39
Error: EigenSparseVectors can only be used in serial!
```
According to [this](https://sourceforge.net/p/libmesh/mailman/message/35173795/):
```
You are attempting to run in parallel, but don't have a parallel
SolverPackage (like PETSc) available for the NumericVector classes.
```

**Comment by davydden** [comment]
+mpi = True by default is good, at the end of the day it's Supercomputer PACKage manager ;-)

**Comment by aprokop** [comment]
Updated PR but adding a PETSC dependency for parallel solvers when `+mpi`.

File: var/spack/repos/builtin/packages/libmesh/package.py [hunk 1]
```diff
@@ -35,8 +35,22 @@ class Libmesh(Package):
 
     version('1.0.0', 'cb464fc63ea0b71b1e69fa3f5d4f93a4')
 
+    variant('mpi', default=True, description='Enables MPI parallelism')
+
+    depends_on('mpi', when='+mpi')
+
+    # Parallel version of libmesh needs parallel solvers
+    depends_on('petsc+mpi', when='+mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmesh/package.py]: Just to confirm, the serial version doesn't need PETSc? It must come with builtin serial solvers.

Comment (aprokop) [inline var/spack/repos/builtin/packages/libmesh/package.py]: I have not tried the serial version but I think it does have builtin solvers. In any case, my changes should not have affected its behavior.

**Pull request merged**